cmd/headscale/cli/utils.go

```diff
@@ -221,10 +221,20 @@ func getHeadscaleConfig() headscale.Config {
 	dnsConfig, baseDomain := GetDNSConfig()
 	derpConfig := GetDERPConfig()
 
+	configuredPrefixes := viper.GetStringSlice("ip_prefixes")
+	prefixes := make([]netaddr.IPPrefix, 0, len(configuredPrefixes))
```

Comment:
So, the effective value of `ip_prefixes` would be `set(ip_prefix ~ ip_prefixes)`? As long as `ip_prefixes` defaults to an empty list, that sounds like a sane, backwards compatible way of handling it, but how does it play along with `ip_prefixes` having a non-zero default value? Wouldn't it mean people who forget to update their configs will unexpectedly see newly joined machines receiving addresses from both their configured pool and the current default IPv4 pool?

Maybe, from a user perspective, it would make more sense to emit a deprecation warning on using `ip_prefix`, and enforce that one or the other *must* be set - but not both at once.

Reply:
Alternatively, do not user `viper` to set the default value, perform the merge+dedup work, and add defaults values if the resulting set is empty - I think that would do the trick.

Reply:
I think that makes sense, I think for a clear message, both emitting a deprecated warning in the logs, but also having it in the `config-example.yaml` make sense, I think there is a surprising amount of users dont reading the logs unless they have to.

I wonder if Viper has a way of marking and handling deprecation?

Reply:
I have implemented this config merging, though I have not tested it manually, yet.

Reply:
Okay, I have tested it, apparently converting the prefix back into a string does not normalize the address by itself, but converting it to a range drops the host bits, so converting *that* back into a prefix, and then a string, allows one to build a canonical representation of them. This way `100.64.0.0/10` and `100.64.0.1/10` maps to the same key, and deduplication happens.

Reply:
Ok thats good, I think that we can live with this until we actually remove the option and can remove that code again.

Reply:
Uff, I forgot a debug println in the last commit.

Reply:
Pushed a fixup.